receiver/hostmetricsreceiver/hostmetrics_receiver_test.go

```diff
@@ -16,36 +16,76 @@ package hostmetricsreceiver
 
 import (
 	"context"
+	"runtime"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector/component/componenttest"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
 	"github.com/open-telemetry/opentelemetry-collector/exporter/exportertest"
 	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver/internal"
+	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver/internal/scraper/cpuscraper"
 )
 
 func TestGatherMetrics_EndToEnd(t *testing.T) {
 	sink := &exportertest.SinkMetricsExporter{}
 
 	config := &Config{
-		DefaultCollectionInterval: 0,
-		Scrapers:                  map[string]internal.Config{},
+		Scrapers: map[string]internal.Config{
+			cpuscraper.TypeStr: &cpuscraper.Config{
+				ConfigSettings: internal.ConfigSettings{CollectionIntervalValue: 100 * time.Millisecond},
+				ReportPerCPU:   true,
+			},
+		},
 	}
 
-	factories := map[string]internal.Factory{}
+	factories := map[string]internal.Factory{
+		cpuscraper.TypeStr: &cpuscraper.Factory{},
+	}
 
 	receiver, err := NewHostMetricsReceiver(context.Background(), zap.NewNop(), config, factories, sink)
+
+	if runtime.GOOS != "windows" {
```

Comment:
Is this an error? The CPU metric seems to work on linux as well, or not?

Reply:
The current code will work on any OS supported by `gopsutil`, but the set of "state" labels / values recorded is limited to ones available in Windows. In the next PR I'm intending to add a bunch more metrics that will only be available on Windows.

I figured it would be better to explicitly restrict this from running on non-Windows operating systems until functionality is properly built out for them rather than potentially confusing users by returning minimal / incomplete data. I don't mind too much either way though.

Reply:
I am confused about what state is missing.

Reply:
See https://github.com/shirou/gopsutil/blob/master/cpu/cpu_linux.go#L251 vs https://github.com/shirou/gopsutil/blob/master/cpu/cpu_windows.go#L160

Linux has other cpu states that are recorded such as Idle, Irq, Softirq, Guest .. that don't necessarily have direct equivalents on Windows (at least not provided by `gopsutil`)

Reply:
Let's do this in a followup PR as discussed.